osdk/src/config_manager/mod.rs

```diff
@@ -104,27 +115,41 @@ fn load_osdk_manifest<S: AsRef<str>>(cargo_args: &CargoArgs, selection: Option<S
             .as_str()
             .unwrap(),
     );
-    let manifest_path = workspace_root.join("OSDK.toml");
 
-    let Ok(contents) = fs::read_to_string(&manifest_path) else {
-        error_msg!(
-            "Cannot read file {}",
-            manifest_path.to_string_lossy().to_string()
-        );
-        process::exit(Errno::GetMetadata as _);
+    // Search for OSDK.toml in the current directory. If not, dive into the workspace root.
+    let manifest_path = PathBuf::from("OSDK.toml");
+    let (contents, manifest_path) = if let Ok(contents) = fs::read_to_string("OSDK.toml") {
+        (contents, manifest_path)
```

Comment:
Now we require that each OSDK-managed crate should have a `OSDK.toml`, right? The `OSDK.toml` should be found from the crate root path, parsed from the result of `cargo metadata`.

Furthur, the rules are missing if the crate's manifest has missing fields. I think the discussion in zulip provides a practical approach:

1. If the crate's manifest has missing values, it will first search for the corresponding option in the workspace's manifest, and then use default values if available.
2. cargo test means cd to the directory of each member crate and then cargo test the individual crate. The manifest in use is determined by Rule 1.
3. cargo run means cd to the directory of the default crate and then cargo run the binary crate. The manifest in use is determined by Rule 1.

Reply:
Then should an empty OSDK.toml exists at every OSDK managed crates in our repo?

Reply:
> Then should an empty OSDK.toml exists at every OSDK managed crates in our repo?

Yes, it's nearly empty, but contains some contents to ensure that this file can be tracked by git. If following the current design, I think the manifest of each crate can contain
```toml
[project]
type = "kernel"
```

Reply:
Also, the default `OSDK.toml` can include empty tables for the most commonly ones. Just like `Cargo.toml` have an empty `dependencies` table.